MINOR:use toList() instead of Collectors.toList()

leaf-soba opened the pull request:
- replace `Stream.collect(Collectors.toList()) ` with `Stream.toList()`

leaf-soba committed:
use toList() instead of Collectors.toList()

replace `Stream.collect(Collectors.toList()) ` with `Stream.toList()`

AndrewJSchofield commented on clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java:
```diff
@@ -457,7 +457,7 @@ public boolean isConnectionSetupTimeout(String id, long now) {
         return connectingNodes.stream()
             .filter(id -> isConnectionSetupTimeout(id, now))
-            .collect(Collectors.toList());
+            .toList();
```
If I understand correctly, `Stream.toList()` was introduced in Java 16. The clients build at Java 11, while the broker builds at Java 17. As a result, I don't think you can use `Stream.toList()` in the clients project.

frankvicky replied:
Yes.
We couldn't invoke `toList()` in modules that still use Java 11.
Please refer to the following:
https://github.com/apache/kafka/blob/c70b7c4b9e6fbd805d9a229b4ac13c1e10016d76/build.gradle#L52

leaf-soba replied:
> If I understand correctly, `Stream.toList()` was introduced in Java 16. The clients build at Java 11, while the broker builds at Java 17. As a result, I don't think you can use `Stream.toList()` in the clients project.

> Yes. We couldn't invoke `toList()` in modules that still use Java 11. Please refer to the following:
> 
> https://github.com/apache/kafka/blob/c70b7c4b9e6fbd805d9a229b4ac13c1e10016d76/build.gradle#L52

Thanks for the clarification! Apologies for not checking the Java version of the modules more carefully. I have reverted the affected modules still using Java 11 to .collect(Collectors.toList());.

frankvicky reviewed (commented):
Thanks for the patch.
I have left a comment.

leaf-soba committed:
revert wrong refactor `toList()`

revert to `Collectors.toList()` in the modules still using Java 11

leaf-soba committed:
delete unnecessary import

remove all unnecessary `import java.util.stream.Collectors;` when using "toList()"

leaf-soba committed:
fix two wrong change import to pass check

fix to pass the spotlessJavaCheck

leaf-soba committed:
revert two cases can't use toList()

List<Object> can't use toList to get the "Object" type

leaf-soba committed:
fix wrong type to pass compileTestJava

fix wrong type to pass compileTestJava

frankvicky commented:
Hi @leaf-soba 
It seems that you are struggling to pass the CI build.
You could execute `./gradlew clean build -x test` locally; it will be more convenient than relying on CI. :)

leaf-soba commented:
> Hi @leaf-soba It seems that you are struggling to pass the CI build. You could execute `./gradlew clean build -x test` locally; it will be more convenient than relying on CI. :)

Thanks, I'll try to run it locally to pass the CI process.

leaf-soba committed:
fix a unittest

fix `testEntityTypes`

ijuma commented:
Note that these methods are not the same, `toList` results in an immutable list. I tried this before and it caused a few tests to fail, so we need to be careful when making this change.

The hard part is validating the cases where the tests do _not_ fail (as there could be missing coverage).

leaf-soba commented:
> Note that these methods are not the same, `toList` results in an immutable list. I tried this before and it caused a few tests to fail, so we need to be careful when making this change.
> 
> The hard part is validating the cases where the tests do _not_ fail (as there could be missing coverage).

@ijuma Thanks for the note! I agree that this PR involves a large change with high risks and minimal benefits to the project. Given the potential issues, I think it would be better to close this PR and create a smaller one that focuses on other refactoring tasks.
